core/src/shred_fetch_stage.rs

```diff
@@ -394,13 +409,15 @@ pub(crate) fn receive_repair_quic_packets(
     }
 }
 
+// Returns true if the feature is effective for the shred slot.
 #[must_use]
-fn should_drop_legacy_shreds(
+fn check_feature_activation(
```

Comment:
could merge with `cluster_nodes::check_feature_activation`

Reply:
That one uses a `root_bank` for less verbosity.
This one uses `feature_set` and `epoch_schedule` because holding onto root bank here has these issues: https://github.com/solana-labs/solana/pull/33078

We can't make `cluster_nodes` one call this one because it adds dependency on `core` crate which we want to avoid.

Reply:
could we make the `cluster_nodes` one also use `feature_set` and `epoch_schedule` so that both callers can use it? 
this code is also duplicated in `duplicate_shred_handler`, but because `cluster_nodes` is in `solana-turbine` it would have added a circular dependency. 
perhaps the best thing would be to move this into `feature_set` so that everyone can use it

Reply:
> could we make the cluster_nodes one also use feature_set and epoch_schedule so that both callers can use it?

The expanded form here is an unfortunate consequence of https://github.com/solana-labs/solana/pull/33078 and is both more verbose and less self-contained; I feel like `cluster_nodes` one taking the root-bank as the argument is already the better code.

> perhaps the best thing would be to move this into feature_set so that everyone can use it

I wouldn't suggest doing that because this one epoch lag is only relevant when working with raw shreds and we don't want to encourage using it in other instances.


Either way, this commit is only renaming existing code and adding a single argument. We can address the code duplication (which pre-exists this commit) separately.
We also need to keep the code change small, because if we are targeting v1.18 for these patches, then this needs to be backported.